File: kothika-frontend/src/api/postService.js
import api from './axiosConfig';

export const postService = {
  getAllPosts: () => api.get('/kothika/posts/getAll'),
  getPostById: (id) => api.get(`/kothika/posts/${id}`),
  createPost: (postRequestDto) => api.post('/kothika/posts/create', postRequestDto),
  updatePost: (id, postRequestDto) => api.put(`/kothika/posts/update/${id}`, postRequestDto),
  deletePost: (id) => api.delete(`/kothika/posts/delete/${id}`),
};

File: kothika-backend/src/main/java/com/sourav/kothika/service/PostService.java
package com.sourav.kothika.service;

import java.util.List;
import java.util.UUID;

import com.sourav.kothika.domain.dto.PostRequestDto;
import com.sourav.kothika.domain.dto.PostResponseDto;

public interface PostService {
	
	public PostResponseDto getPostById(UUID id);
	public List<PostResponseDto> getAllPosts();
	public List<PostResponseDto> getAllPublishedPostsOfUser();
	public List<PostResponseDto> getAllDraftPostsOfUser();
	public PostResponseDto createPost(PostRequestDto postRequestDto);
	public String deletePost(UUID id);
	public PostResponseDto updatePost(UUID id, PostRequestDto postRequestDto);
}
